MathLib/LinAlg/SimpleMatrixProvider.h

```diff
@@ -0,0 +1,90 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATHLIB_SIMPLE_MATRIX_PROVIDER_H
+#define MATHLIB_SIMPLE_MATRIX_PROVIDER_H
+
+#include<map>
+#include<memory>
+
+#include "MatrixProviderUser.h"
+
+namespace MathLib
+{
+
+template<typename Matrix, typename Vector>
+class SimpleMatrixProvider final
```

Comment:
It provides both matrix and vector. As I commented before, the class name doe not give the full meaning of the class.